File: agent/commands-generator/src/main/java/org/matilda/commands/python/writer/PythonClass.java
package org.matilda.commands.python.writer;

public class PythonClass extends PythonCodeBlock {
    private static final int CLASS_EMPTY_LINES = 1;

    PythonClass(CodeBlock codeBlock) {
        super(codeBlock, CLASS_EMPTY_LINES);
    }

    public PythonCodeBlock addInstanceMethod(PythonFunctionSpec functionSpec) {
        PythonFunctionSpec methodSpec = functionSpec.copyBuilder().addParameterAtStart("self").build();
        return newFunction(methodSpec);
    }
}

File: agent/commands-generator/src/main/java/org/matilda/commands/python/writer/PythonClass.kt
package org.matilda.commands.python.writer

class PythonClass internal constructor(codeBlock: CodeBlock) : PythonCodeBlock(codeBlock, CLASS_EMPTY_LINES) {
    fun addInstanceMethod(functionSpec: PythonFunctionSpec): PythonCodeBlock {
        val methodSpec = functionSpec.copyBuilder().addParameterAtStart("self").build()
        return newFunction(methodSpec)
    }

    fun addStaticMethod(functionSpec: PythonFunctionSpec): PythonCodeBlock {
        val methodSpec = functionSpec.copyBuilder().addAnnotation("staticmethod").build()
        return newFunction(methodSpec)
    }

    companion object {
        private const val CLASS_EMPTY_LINES = 1
    }
}
